internal/app/store/url_store.go

```diff
@@ -24,7 +36,7 @@ func (s *URLStore) SetURL(key, value string) error {
 		return ErrEmptyValue
 	}
 	s.urls[key] = value
-	return nil
+	return CreateSnapshot(s)
```

Comment:
[CHANGE REQUEST] We should do it only once at the end of the program's execution.